File: 06C/ex1_6.c
#include <stdio.h>

int main(int argc, char *argv[])
{
    int initialDivisor = 5;
    int totalIterations = 0;

    for (int i = 1; i <= initialDivisor; i++) {
    
        int dividend = i;
        int divisor = initialDivisor;
        int iterationsToSolveItCnt = 0;

        do  {
            iterationsToSolveItCnt++;
            int remainder = dividend % divisor;
            if (remainder == 0) break;
            dividend = divisor;
            divisor = remainder;
        } while (1==1);

        totalIterations += iterationsToSolveItCnt;
    }

    float result = (float)totalIterations / initialDivisor;

    printf("%f", result);
    return 0;
}

File: 02Python/ex1_6.py
"""What is T5, the avg # of times step E1 is performed (iterations) when n = 5?"""
def main():
    """Execute Euclid's algorithm for n=5 count the iterations and average"""
    initial_divisor = 5
    total_iterations = 0

    for index in range(1, initial_divisor + 1):
        dividend = index
        divisor = initial_divisor
        iterations_to_solve_cnt = 0

        while True:
            iterations_to_solve_cnt += 1
            remainder = dividend % divisor
            if remainder == 0:
                break
            dividend = divisor
            divisor = remainder

        total_iterations += iterations_to_solve_cnt

    result = float(total_iterations) / initial_divisor

    print(result)


if __name__ == "__main__":
    main()
